libs/wire-subsystems/src/Wire/ConversationStore/Cassandra.hs

```diff
@@ -578,8 +580,15 @@ removeRemoteMembersFromLocalConv cnv victims = do
 members :: ConvId -> Client [LocalMember]
 members conv = do
   parents <- retry x1 $ query Cql.selectConvParent (params LocalQuorum (Identity conv))
-  concatMap (nubBy ((==) `on` (.id_)) . mapMaybe toMember)
-    <$> UnliftIO.pooledMapConcurrentlyN 16 fetchMembers (conv : mapMaybe runIdentity parents)
+  -- map (nubBy ((==) `on` (.id_)) . mapMaybe toMember . map (either id id))
+  nubBy ((==) `on` (.id_))
+    . map (either id id)
+    . sortBy (liftCompare2 (comparing (.id_)) (comparing (.id_)))
+    . concatMap (bitraverse (mapMaybe toMember) (mapMaybe toMember)) -- put shadow conversations members, 'conv', first to use it in priority, when there are two
+    <$> UnliftIO.pooledMapConcurrentlyN
+      16
+      (bitraverse fetchMembers fetchMembers)
+      (Left conv : mapMaybe (fmap Right . runIdentity) parents)
```

Comment:
The problem with the old code was not that `pooledMapConcurrentlyN` reorders things, its that we do `nubBy` inside the `concatMap`. So, I think its easier if we implement this like this:

```haskell
nubBy ((==) `on` (.id_)) . concatMap (mapMaybe toMember)
    <$> UnliftIO.pooledMapConcurrentlyN 16 fetchMembers (conv : mapMaybe runIdentity parents)
```

Reply:
@akshaymankar great, it works so far, thanks a lot!